example.cc

```diff
@@ -22,10 +22,10 @@ int main() {
 
   // Fill with random values in range [-2, 2].
   srand(1);
-  for (Index i = 0; i < A_rows * width; ++i) {
+  for (Index i = 0; i < A.size(); ++i) {
     A[i] = ((float)rand()/(float)RAND_MAX)*4.0f - 2.0f;
   }
-  for (Index i = 0; i < width * B_cols; ++i) {
+  for (Index i = 0; i < B.size(); ++i) {
```

Comment:
At this rate you may as well do `for (float *b : B)`

Reply:
You're right. I see the pull request is merged so I'll change it next time.